telegram/ext/_dispatcher.py

```diff
@@ -198,10 +218,17 @@ def __init__(
                 stacklevel=stack_level,
             )
 
-        self.user_data: DefaultDict[int, UD] = defaultdict(self.context_types.user_data)
-        self.chat_data: DefaultDict[int, CD] = defaultdict(self.context_types.chat_data)
+        self._user_data: DefaultDict[int, UD]
+        self._chat_data: DefaultDict[int, CD]
+        # Read only mapping-
+        self.user_data: Mapping[int, UD]
+        self.chat_data: Mapping[int, CD]
+
+        self._set_user_data(defaultdict(self.context_types.user_data))
+        self._set_chat_data(defaultdict(self.context_types.chat_data))
```

Comment:
did we decide on switching to using to `_chat._data.update(…)` instead of `_set_chat_data(…)`? I lost track :D

Reply:
The `.update()` thing was planned for another PR, which will also update the return types of `get_user/chat/_data`

Reply:
Ah, I was unclear again 😬  I meant that adjusting the documented return type of `get_*_data` and the implementations of `PP/DictPersistence` can be done in another PR. switching to `.update()` should work without that (since defaultdicts are dicts) and adding `_set_*_data` here just to remove it in the next PR again seems a bit superfluous to me